src/hooks/useCatalogResults/useCatalogResults.ts

```diff
@@ -15,31 +15,38 @@ export interface UseCatalogResultsOptions {
  */
 export const useCatalogResults = ({
   limit,
-  offset,
-  query,
-  language,
+  offset = 0,
+  query = "",
+  language = null,
 }: UseCatalogResultsOptions) => {
   const { data, loading, error } = useCatalog();
 
   const results = useMemo(() => {
     if (loading || error || !data?.packages) return [];
 
-    let filtered = data.packages;
+    const filtered = data.packages.filter((item) => {
+      // Filter out items which do not support the current language
+      if (language && language !== Language.TypeScript) {
+        if (!Object.keys(item.languages).includes(language)) {
+          return false;
+        }
+      }
+
+      // Filter items which do not match the current query
+      if (query.length) {
+        const itemStr = JSON.stringify(item).toLowerCase();
+        const queryStr = query.toLowerCase();
 
-    if (query.length) {
-      filtered = filtered.filter((item) =>
-        JSON.stringify(item).toLowerCase().includes(query.toLowerCase())
-      );
-    }
+        if (!itemStr.includes(queryStr)) {
+          return false;
+        }
+      }
 
-    // Omit TS from filters b/c it is always available
-    if (language && language !== Language.TypeScript) {
-      filtered = filtered.filter((item) =>
-        Object.keys(item.languages).includes(language)
-      );
-    }
+      return true;
+    });
 
-    return filtered.sort((p1, p2) => {
+    // Next sort by most recently updated. (The sort should eventually become a parameter)
+    return [...filtered].sort((p1, p2) => {
```

Comment:
what is the need to copy the array over before sorting?

Reply:
to avoid mutating the original dataset - this actually was causing the "bug" that was allowing tests to pass in the first place